src/sst/core/params.h

```diff
@@ -616,6 +616,65 @@ class Params : public SST::Core::Serialization::serializable
     static void
     insert_global(const std::string& set, const key_type& key, const key_type& value, bool overwrite = true);
 
+
```

Comment:
I think it's dangerous to put functions in a public place and then expect people not to call them.  If I'm using an editor with completion I might just see a method `getGlobalParamSet(std::string const&) -> std::map<std::string, std::string>` and not realize that you didn't intend for this to be called by users.

Reply:
Yea, this is a tricky issue.  Technically, the whole configuration interface (including Params) needs to be public because you can implement inputter and outputter classes in element libraries (or will soon be able to).  However, that doesn't mean they should be used by Component/SubComponent.  I'm afraid there's no way to differentiate that other than documentation.  For now, I'm not claiming backward compatibility and scourging people from using them because they could still in flux.  Once they've stabilized and all the APIs for writing your own inputter/outputter in you element library are finalized, the comments will change to reflect where they should/should not be used.

Reply:
Can't you just do 
```c++
namespace detail {
 static void insert_global(...);
}
```
It doesn't stop someone from calling it, but the extra namespace make it's clear this is Core guts and isn't required to remain compatible or be stable.

Reply:
Yes, but then when I finalize the interface I have to go back and remove the detail namespace from everything.  Remember, this does need to be public outside of core once it's finalized and a few other pieces are in place.